src/test/resources/com/puppycrawl/tools/checkstyle/checks/blocks/InputRightCurlyBracePolicyAloneForAnonymousClass.java

```diff
@@ -0,0 +1,107 @@
+package com.puppycrawl.tools.checkstyle.checks.blocks;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Set;
+import java.util.TreeSet;
+import java.util.stream.Collectors;
+
+class InputRightCurlyBracePolicyAloneForAnonymousClass {
+
+    private static final Set<String> TREE_SET = Collections.unmodifiableSortedSet(
+            Arrays.stream(new String[] {"br", "li", "dt", "dd", "hr", "img", "p", "td", "tr", "th", })
+                    .collect(Collectors.toCollection(TreeSet::new))); // for coverage
+    public static final String STRING = new String("String");
+    public static final Object OBJECT1 = new Object();
+    public static final Object OBJECT2 = new Object() {
+        @Override
+        public String toString() {
+            return "";
+        }
+    }; int field1;   // violation
+
+    int field2;
+
+
+    static void method1(Object o) {
+        new Object() { @Override public String toString() { return ""; } int binary = 10; }; // violation
+    }
+
+    static void method2(Thread t, int x) {
+        method1(new Object() {
+            @Override
+            protected Object clone() {
+                return new Object() { @Override protected void finalize() { "".toString(); } }; // violation
+            }
+        }); // no violation
```

Comment:
@romani @rnveach 
This is a controversial example. What do you think?

Reply:
@romani @MEZk @rnveach 
This Point was left out. Regression reports for option _alone_ and _alone-or-singleline_ have been approved. There is still no violation here. Please confirm if a violation is needed here.

Reply:
Done. There is a violation for this now.